app/services/transaction_search/order_status_searcher.rb

```diff
@@ -5,8 +5,8 @@ class OrderStatusSearcher < BaseSearcher
     def options
       # Unlike the other lookups, this query is much faster this way than using a subquery
       OrderStatus.find_by_sql(order_details.joins(:order_status)
-                                                         .select("distinct(order_statuses.id), order_statuses.facility_id, order_statuses.name, order_statuses.lft")
-                                                         .reorder("order_statuses.lft").to_sql)
+                                           .select("distinct(order_statuses.id), order_statuses.facility_id, order_statuses.name, order_statuses.lft")
+                                           .reorder("order_statuses.lft").to_sql).uniq
```

Comment:
I feel like the `uniq` shouldn't be necessary. What caused it to become necessary? The `.select("distinct(order_statuses.id)` should have it covered.

Reply:
I'm confused by that too, but for some reason in the new/in process tabs we were getting multiple (literally as many as there were order details) instances of "new" and "in process" appearing as options. Weirdly, when I tried to filter by one of the "new"s, it would do the filter correctly (find 22 of 22 new order details), but then the selector would have all 22 "new"s shown as selected.

I ended up throwing the uniq on it because I figured that wouldn't break anything and it didn't seem worth doing a deep dive into why I was getting those results when it was behaving as expected in the billing tab, but if it bothers you excessively I can look into it more.

Reply:
It only concerns me if the query itself returns an OrderStatus object for _every single_ order detail, and then it gets `uniq`ed. If that's thousands of order details (especially if it's not paginated), that could balloon memory. If the `uniq` changes the SQL, then it's probably fine.

Reply:
Ah, it definitely is returning an OrderStatus object for every single order detail. They're distinct objects too, not just the same one multiple times. I'll play around with restructuring the query and see if there's a better way to do this. I don't get why the distinct isn't working...